docs/source/monitoring/metrics.md

```diff
@@ -139,35 +139,43 @@ Specifying this function overrides the [`defaultGenerateClientInfo` function](ht
 
 ## Logging
 
-Apollo Server provides two ways to log a server: per input, response, and errors or periodically throughout a request's lifecycle. Treating the GraphQL execution as a black box by logging the inputs and outputs of the system allows developers to diagnose issues quickly without being mired by lower level logs. Once a problem has been found at a high level, the lower level logs enable accurate tracing of how a request was handled.
+You can set up fine-grained operation logging in Apollo Server by defining a custom **plugin**. Apollo Server plugins enable you to perform actions in response to individual phases of the GraphQL request lifecycle, such as whenever a GraphQL request is received from a client.
 
-### High-level logging
+The example below defines a plugin that responds to three different operation events. As it shows, you provide an array of your defined `plugins` to the `ApolloServer` constructor.
 
-Apollo Server allows `formatError` and `formatResponse` configuration options which can be defined as callback-functions which receive `error` or `response` arguments respectively.
-
-For the sake of simplicity, these examples use `console.log` to output error and debugging information though a more complete example might utilize existing logging or error-reporting facilities.
+For a list of available lifecycle events and their descriptions, see [Plugins](../integrations/plugins/).
 
 ```js
+const myPlugin = {
+
+  // Fires whenever a GraphQL request is received from a client.
+  requestDidStart(requestContext) {
+    console.log(`Request started! Query string:
+    ${requestContext.request.query}`);
```

Comment:
Depending on the query (particularly if it's multi-lined or not, this will print in a somewhat peculiar manner in the log because of the four-spaces of indentation on the second line.  

Similarly, it looks a bit off to me in this snippet because it's not indented under its opening `console` line.  I might suggest something like:

```js
    console.log("Request started! Query\n" +
      requestContext.request.query});
```

So it would be flush-left in the output.

Reply:
Ah thanks, will do! Thought I was being clever with a template string. Nope.